clang/lib/Driver/ToolChains/HLSL.cpp

```diff
@@ -226,6 +226,47 @@ HLSLToolChain::TranslateArgs(const DerivedArgList &Args, StringRef BoundArch,
       A->claim();
       continue;
     }
+    if (A->getOption().getID() == options::OPT_dxc_hlsl_version) {
+      // Translate -HV into -std for llvm
+      // depending on the value given
+      llvm::StringRef value(A->getValue());
+      llvm::StringRef acceptedValues(
+          Opts.getOptionValues(options::OPT_dxc_hlsl_version));
+      llvm::SmallVector<StringRef, 8> validValues;
+      acceptedValues.split(validValues, ", ");
+      if (value == validValues[0]) {
```

Comment:
Why not just check value == "2016" here so we don't need to getOptionValues?
And the code will be easier to understand.

Reply:
The idea to use getOptionValues was an idea inspired by @llvm-beanz 's feedback, to prevent littering direct string comparisons into the code and unify string handling. Originally, I did have the direct "2016" comparison. I'm not sure if the LLVM community would want access to these option values, but it seems like the most convenient way to get around direct string comparisons. I do remember string comparisons were a pain point in the DXC codebase and want to try and avoid that.

Reply:
We can do something like https://github.com/llvm/llvm-project/blob/main/clang/lib/Frontend/CompilerInvocation.cpp#L3689 where the OPT_std_EQ is checked 
 with the LangStandard::getLangKind at https://github.com/llvm/llvm-project/blob/main/clang/lib/Basic/LangStandards.cpp#L64

Maybe add LangStandard::getHLSLLangKind ?

Reply:
I've added your recommended function, hopefully it is what you envisioned?